### DISCUSSION TITLE
Request for three new sensors

### DISCUSSION THREAD

**Discussion by martindell** [Arnold-n/P1P2MQTT#54] [Ideas]
My P1P2 interface has been working without any problems for a few months now (thank you Arnold).  I'm just using MQTT actions to trigger heat ON/OFF and DHW ON/OFF in a few scripts and automations.  

My Daikin indoor unit is: EHBH16CAV3 and I wonder if there's a way of exposing three new sensors in the P1P2 interface:

- BUH (actuators)
- Defrost / oil return (operation modes)
- 3-way valve (actuators)

I can see these values in the Daikin display under Information > Actuators and Operation modes - so they're in there somewhere!

**Comment by Arnold-n** [comment]
It is easy to expose these sensors when we know which bits/bytes represent these values - can you match any particular bits or bytes to the changes in the user menu by logging P1P2/R/# during such a value change (or share the log)?

**Reply by martindell** [reply]
I'll have a try ..

**Comment by martindell** [comment]
This log should contain actions for the BUH and 3-way valve - unsure how to watch for 'Defrost' at the moment.  Sorry, but I don't know how to trim it for the correct time period.

[230102225350.log.zip](https://github.com/Arnold-n/P1P2Serial/files/10334965/230102225350.log.zip)

I guesstimate 'Backup' from power consumption at the moment, so around 02:01 and 02:32 in the logs should show something:

<img width="488" alt="backup power actuator" src="https://user-images.githubusercontent.com/1224738/210319057-f2b2424a-703b-4948-a9dd-76bdd096dd04.png">

DHW, and therefore the 3-way valve (without backup or booster heaters) kicked in at 00:30 and would have ended when the pump switched off at 02:06

<img width="1029" alt="pump and dhw actuators" src="https://user-images.githubusercontent.com/1224738/210319280-4ac07723-c5be-400c-83a6-b4922f025a5f.png">

**Comment by martindell** [comment]
For the complete picture, here's the status of the relevant sensors this morning.  I have no idea what S1_Gasboiler_on is:

<img width="1171" alt="Screenshot 2023-01-03 at 08 32 11" src="https://user-images.githubusercontent.com/1224738/210322786-801bf421-e27c-420c-868d-102f8d35bff4.png">

Maybe the three refrigerant temperature sensors give a clue when defrost is happening?

<img width="1166" alt="Screenshot 2023-01-03 at 08 39 40" src="https://user-images.githubusercontent.com/1224738/210323773-0feb7979-9240-4570-bd6e-4d77eca1fa65.png">

**Comment by martindell** [comment]
I wonder if the sensor S1_DHW_Active_1 is in fact the three-way valve?  Seems consistent with pump / backup & booster heater sensor timings?

<img width="1170" alt="Screenshot 2023-01-03 at 13 55 45" src="https://user-images.githubusercontent.com/1224738/210371269-4f36f681-8363-4618-b18b-de812c603472.png">

**Comment by Arnold-n** [comment]
Thanks @martindell, threre are a lot of changes in your datalog, most of these got a name (some perhaps wrong), not all of them visibile in HA, let me put them in a time line (name at end of line). I think that BUH = 400010 byte 18 bit 4, and defrost was already known as 400010 byte 17 bit 1. I think the Valve_3Way is tank-related but am not sure about the interaction with the Valve_DHW_Tank and Valve_Zone_Main:

```
DHW-on/off by controller
000010:2:0      0:29a--(DHW-on?)-----------------------2:32a                                   7:32a---7:32c ("DHW_OnOff")
000010:17:6     0:29a--(DHW-on?)-----------------------2:32a                                   7:32a---7:32c ("DHW_Related_Q")
(and 00:29c setpoint changed from 65 to 55)
DHW-related:
400010:17:6     0:29a----------(DHW?)------------------2:32a                                   7:32a---7:32c
400010:3:0        0:29b--------(DHW?)------------------2:32a                                     7:32b-7:32c ("Valve_3Way")
BUH:
400010:18:4                       1:57-1:57 2:01-(BUH?)--2:32b
Defrosts?:
400010:17:1         0:42--0:45                                        3:09-3:10b 6:57-6:58                   ("Defrost_Operation")
Other:
400010:0:0                                                   2:32d---------------------------------------->  ("Heating_On_Off_1")
400010:2:5  <-----------0:44b 0:48-----------------2:05      2:32d------3:10a 4:51------------------------>  ("Valve_Zone_Main")
400010:2:7              0:44b-----------------2:02 2:05--2:32b                                               ("Valve_DHW_Tank")
400012:9:2  23:48a--------------1:07                 2:07-----------2:57      4:51------------------------>
400012:9:7            0:44a-------------------2:02 
400012:11:7 <---------0:44a   0:48-----------------2:05          2:44---3:10a
400012:12:6   23:48b----0:44b                                  2:37-2:57        4:57----7:05 7:13--------->  ("Gasboiler_on")
400012:12:7             0:44b------------------------------2:32c                                             ("DHW_Active_1")
```

**Comment by martindell** [comment]
Oh nice work - thanks for sorting through all that Arnold!  
I'm not sure how to make / monitor HA sensors from these addresses though? 
Also looks like there are new sensors compared to the firmware I'm running? 
My P1P2-bridge-esp8266 is 0.9.17 and P1P2Monitor is 0.9.17-fix1. What versions should I flash to test these sensors?

**Comment by Arnold-n** [comment]
Some parameters like Defrost were recently added; the code starts at [this line](https://github.com/Arnold-n/P1P2Serial/blob/main/examples/P1P2-bridge-esp8266/P1P2_Daikin_ParameterConversion_EHYHB.h#L1579) and if you add HACONFIG to a line the relevant parameter will be visible in HA. But HA already has some issues with the high influx of data. 

I'll upload new images later when I have time.
